File: fetch_title_articles/articles.py
import requests

BASE_URL = "https://jsonmock.hackerrank.com/api/articles"


def get_articles(author_name):
    article = []
    page = 1

    while True:
        try:
            # params
            params = {"page": page}
            if author_name is not None:
                params["author"] = author_name

            response = requests.get(BASE_URL, params=params)
            data = response.json().get('data', [])
            if not data:
                break

            article.extend(data)
            page += 1
        except Exception as e:
            print("something happens when fetching data: ", str(e))
            break

    return article

def get_title_articles(article):
    title_article = list(
        set(
            article["title"] or article["story_title"]
            for article in article
            if article["title"] or article["story_title"]
        )
    )
    return sorted(title_article)

def execute():
    author_name = input("")
    articles = get_articles(author_name)
    article_names = get_title_articles(articles)
    print("\n".join(article_names))

if __name__ == "__main__":
    execute()

File: fetch_title_articles/articles.js
const BASE_URL = "https://jsonmock.hackerrank.com/api/articles";


const getArticles = async (author) => {
    let articles = [];
    let page = 1;

    while (true) {
        try {
            const params = new URLSearchParams({page: page});
            if (author !== null || author !== "") {
                params.append("author", author)
            }

            const response = await fetch(`${BASE_URL}?${params.toString()}`);
            const data = (await response.json())?.data || [];

            if (data.length === 0) {
                break
            }

            articles.push(...data);
            page++;
        } catch (e) {
            console.error("Something happens when fetch data: ", e);
            break;
        }
    }

    return articles;
}

const getTitleArticles = async (articles) => {
    const titles = [...new Set(
        articles.map(article => article?.title || article?.story_title).filter(title => title)
    )];
    return titles.sort();
}

const execute = async () => {
    try {
        const readline = require('readline');
        const rl = readline.createInterface({ input: process.stdin, output: process.stdout});

        rl.question("", async (authorName) => {
            const articles = await getArticles(authorName.trim() || null);
            const articleNames = await getTitleArticles(articles);
            console.log(articleNames.join("\n"));
            rl.close();
        });
    } catch (error) {
        console.error("Error:", error);
    }
}

execute();